src/modules/setup/index.ts

```diff
@@ -1,14 +1,44 @@
-/* eslint-disable @typescript-eslint/no-unused-vars */
+import chalk from 'chalk'
+import logger from '../../logger'
 import { getManifest } from '../../manifest'
 import { setupTooling } from './setupTooling'
 import { setupTSConfig } from './setupTSConfig'
 import { setupTypings } from './setupTypings'
 
-export default async (opts: { i?: boolean; 'ignore-linked': boolean }) => {
+interface SetupOpts {
+  i?: boolean
+  'ignore-linked'?: boolean
+  all?: boolean
+  tooling?: boolean
+  typings?: boolean
+  tsconfig?: boolean
+}
+
+export default async (opts: SetupOpts) => {
+  const { tooling, typings, tsconfig } = opts
```

Comment:
maybe it would be better to define the `tooling`, `typings` and `tsconfig` variables the same way you to for `all`, so you don't need to do the check `something || all`, like

```ts
const tooling = opts.tooling || all
const typings = opts.typings || all
const tsconfig = opts.tsconfig || all
```

would be nice if the CLI framework already did this for us :-/